### ISSUE TITLE
Hatchery: Add the ability to order filters

### ISSUE THREAD

**Issue by Farigh** [Farigh/pokeclicker-automation#130]
### Is your feature request related to a problem? Please describe.
For now, if a user set multiple filters, it will always proceed in a predetermined order.

For example, setting : 
  - Region: Kanto
  - Type: Fire
  - Shinies last

Will behave like this : 
  - Hatch pokemon matching every criteria
  - If there is no more Kanto Fire non-shinies pokémon, hatch Kanto Fire shinies
  - If there is no more Kanto Fire pokémon, hatch Kanto non-Fire non-shinies
  - ...

In short, Region will always prevail over Type, that will always prevail over non-Shinies

### Describe the solution you'd like

The player should be able to choose the priorisation order.
The ideal implementation would be to be able to drag and drop priorisation settings to re-order them.

**Comment by linkdest** [comment]
Going to use this issue as it's the same topic.

The problem is that I'm at end of unova and only hatch water, fairy and toxicroak. The new option is nice, but it end up using pokemons with low effiency.

Suggestion: Only hatch pokemons that are categorized, can be only favorites. To avoid always the same pokemons being hatched, an option would be have a list order the same as the game: breed effiency, times hatched and attack.
